discord/ext/menus/__init__.py

```diff
@@ -579,7 +579,10 @@ async def _internal_loop(self):
                 # consider this my warning.
 
         except asyncio.TimeoutError:
-            pass
+            try:
+                await self.timed_out()
+            except Exception:
+                pass
```

Comment:
```suggestion
            self.__timed_out = True
```

I think rather than a new method `finalize` should be called with a new argument (`timed_out`) in the `finalize` block which is passed `self.__timed_out` and then cleared with `self.__timed_out = False` both in the beginning of the block and after `finalize` is called inside `finally`.

Of course the documentation has to be updated to reflect these changes.